ocfl-java-core/src/main/java/edu/wisc/library/ocfl/core/extension/storage/layout/NTupleOmitPrefixStorageLayoutExtension.java

```diff
@@ -0,0 +1,168 @@
+
+/**
+ * Implementation of the <a href="https://ocfl.github.io/extensions/0007-n-tuple-omit-prefix-storage-layout.md">
+ * N Tuple Storage Layout</a> extension.
+ *
+ * @author vcrema
+ */
+public class NTupleOmitPrefixStorageLayoutExtension implements OcflStorageLayoutExtension {
+
+    public static final String EXTENSION_NAME = "0007-n-tuple-omit-prefix-storage-layout";
+
+    private static final Logger LOG = LoggerFactory.getLogger(NTupleOmitPrefixStorageLayoutExtension.class);
+    
+    private NTupleOmitPrefixStorageLayoutConfig config;
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getExtensionName() {
+        return EXTENSION_NAME;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getDescription() {
+        return "This storage root extension describes an OCFL storage layout "
+        		+ "combining a pairtree-like root directory structure derived from "
+        		+ "prefix-omitted object identifiers, followed by the prefix-omitted "
+        		+ "object identifier themselves. The OCFL object identifiers are "
+        		+ "expected to contain prefixes which are removed in the mapping to "
+        		+ "directory names. The OCFL object identifier prefix is defined as "
+        		+ "all characters before and including a configurable delimiter. "
+        		+ "Where the prefix-omitted identifier length is less than "
+        		+ "tuple size * number of tuples, the remaining object id (prefix omitted) "
+        		+ "is left or right-side, zero-padded (configurable, left default), "
+        		+ "or not padded (none), and optionally reversed (default false). "
+        		+ "The object id is then divided into N n-tuple segments, and used "
+        		+ "to create nested paths under the OCFL storage root, followed by "
+        		+ "the prefix-omitted object id directory.";
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public synchronized void init(OcflExtensionConfig config) {
+    	Enforce.notNull(config, "configFile cannot be null");
+        
+    	if (!(config instanceof NTupleOmitPrefixStorageLayoutConfig)) {
+    		throw new OcflExtensionException(String.format("This extension only supports %s configuration. Received: %s",
+             	getExtensionConfigClass(), config));
+     	}
+
+    	NTupleOmitPrefixStorageLayoutConfig castConfig = (NTupleOmitPrefixStorageLayoutConfig) config;
+
+     	this.config = castConfig;
+    }
+
+
+    @Override
+    public Class<? extends OcflExtensionConfig> getExtensionConfigClass() {
+        return NTupleOmitPrefixStorageLayoutConfig.class;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String mapObjectId(String objectId) {
+        if (config == null) {
+            throw new OcflExtensionException("This extension must be initialized before it can be used.");
+        }
+        
+        if (!objectId.matches("\\A\\p{ASCII}*\\z")) {
```

Comment:
I like this, but please compile it to a static constant `Pattern` at the class level so that the regex does not need to be recompiled every time this method is called.